src/js/app.js

```diff
@@ -0,0 +1,151 @@
+import { calculatorInitValue, calculatorSymbols, calculatorDigitMaxLength } from './constants.js';
+import { calculate } from './calculator.js';
+
+class App {
+  /** @type {HTMLDivElement} */
+  #element;
+  /** @type {HTMLHeadingElement} */
+  #totalElement;
+  /** @type {boolean} */
+  #isCalculateAfter = false;
+
+  /**
+   * @param {HTMLDivElement} element
+   */
+  constructor(element) {
+    this.#element = element;
+    this.#render();
+    this.#mounted();
+  }
+
+  /**
+   * @param {string} value
+   */
+  calculatorInput(value) {
+    const total = this.#totalElement.textContent;
+    const regExp = /([0-9]+)/g;
+    const totalLastNumbers = total.match(regExp).pop();
+    const isLastStringContainsSymbol = calculatorSymbols.includes(total.slice(-1));
+
+    // 심볼을 입력한 경우
+    if (calculatorSymbols.includes(value)) {
+      this.#totalElement.textContent = isLastStringContainsSymbol
+        ? total.substr(0, total.length - 1) + value
+        : total + value;
+      return;
+    }
+
+    // 번호를 입력했지만 번호가 최대 자리수를 넘치지 않은 경우
+    if (totalLastNumbers.length < calculatorDigitMaxLength || isLastStringContainsSymbol) {
+      this.#totalElement.textContent = total === calculatorInitValue ? value : total + value;
+      return;
+    } else {
+      throw new Error(`번호를 ${calculatorDigitMaxLength}보다 길게 작성할 수 없습니다.`);
+    }
+  }
+
+  /**
+   * @param {MouseEvent} e
+   */
+  #handleDigitsClick = (e) => {
+    /**
+     * @typedef {object} DestructuredEvent
+     * @property {Element} target
+     */
+    /** @type {DestructuredEvent} */
+    const { target } = e;
+
+    if (target.classList.contains('digit') && this.#isCalculateAfter === false) {
+      const digit = target.textContent;
+      this.calculatorInput(digit);
+    } else if (this.#isCalculateAfter) {
+      throw new Error('계산를 한 뒤에는 바로 번호를 입력할 수 없습니다.');
+    }
+  };
+
+  /**
+   * @param {MouseEvent} e
+   */
+  #handleModifiersClick = (e) => {
```

Comment:
구조분해할당을 아래와 같이 바로 사용하면 가독성이 더 좋아질거 같아요!
```suggestion
  #handleModifiersClick = ({ target }) => {
```